test/dotnet-compile.Tests/CompilerTests.cs

```diff
@@ -167,6 +167,24 @@ public void DepsDependencyContextIsValidOnBuild()
                 runCommand.Execute().Should().Pass();
         }
 
+        [Fact]
+        public void CanSetOutputAssemblyName()
+        {
+            var testInstance =
+                TestAssetsManager
+                    .CreateTestInstance("LibraryWithOutputAssemblyName")
+                    .WithLockFiles();
+
+            var root = testInstance.TestRoot;
+            var outputDir = Path.Combine(root, "bin");
+            var testProject = ProjectUtils.GetProjectJson(root, "LibraryWithOutputAssemblyName");
+            var buildCommand = new BuildCommand(testProject, output: outputDir, framework: DefaultFramework);
+            var result = buildCommand.ExecuteWithCapturedOutput();
+            result.Should().Pass();
+
+            new DirectoryInfo(outputDir).Should().HaveFiles(new [] { "MyLibrary.dll" });
```

Comment:
@pakrym I tried it and I can see that deps now has: 
```
        "runtime": {
          "MyApp.dll": {}
        }
```
Is this what needs to be checked, right?

(on a separate note when creating an app the dll is named as per the requested output name but the .exe itself has the package name - I think we want .exe to have the same name as specified in the outputName right @davidfowl ? if so, I will investigate this separately...)